src/Agent.Listener/Configuration/LinuxServiceControlManager.cs

```diff
@@ -241,7 +241,7 @@ private void ExecuteSystemdCommand(string command)
                 // TODO: can systemd installed in non default directory?
                 using (var cs = new CancellationTokenSource(TimeSpan.FromSeconds(45)))
                 {
-                    processInvoker.ExecuteAsync("/usr/bin", "systemctl", command, null, cs.Token).Wait();
+                    processInvoker.ExecuteAsync("/usr/bin", "systemctl", command, null, cs.Token, Timeout.Infinite).Wait();
```

Comment:
Instead of changing all this code can't we have an overload in process invoker or default to infinite?